src/server/api/url_parser.cpp

```diff
@@ -1,88 +1,78 @@
 #include "server/api/url_parser.hpp"
-
 #include "engine/polyline_compressor.hpp"
 
-#include <boost/bind.hpp>
-#include <boost/spirit/include/qi_char_.hpp>
-#include <boost/spirit/include/qi_grammar.hpp>
-#include <boost/spirit/include/qi_uint.hpp>
-#include <boost/spirit/include/qi_real.hpp>
-#include <boost/spirit/include/qi_lit.hpp>
-#include <boost/spirit/include/qi_action.hpp>
-#include <boost/spirit/include/qi_as_string.hpp>
-#include <boost/spirit/include/qi_operator.hpp>
-#include <boost/spirit/include/qi_plus.hpp>
```

Comment:
Great to hear you're getting the same results. Spirit v2 is really hard for getting above a certain level of understanding, it's almost as if I have to learn by failing. Had some hard times the last two days.

I took a look at x3 for a few side-projects and their approach to always generate an AST and move away from semantic actions and grammar is refreshing.